### ISSUE TITLE
Sprites with off-center .center properties potentially not rendered at frustrum edges.

### ISSUE THREAD

**Issue by LukasWho156** [mrdoob/three.js#24822]
For example, if a Sprite's center is set to (0, 0.5) and the sprite moves towards the left edge of the camera frustrum, it will disappear as soon as less than ~ 30% of the sprites remain.

Simple code example:

```
const renderer = new THREE.WebGLRenderer();
renderer.setSize(400, 300);
document.body.append(renderer.domElement);

const scene = new THREE.Scene();
const sprite = new THREE.Sprite(new THREE.SpriteMaterial());
sprite.scale.set(100, 100, 1);
sprite.center = new THREE.Vector2(0, 0.5);
scene.add(sprite);

const camera = new THREE.OrthographicCamera(-200, 200, 150, -150, 0.1, 1000);
//const camera = new THREE.PerspectiveCamera(50, 4/3, 0.1, 1000);
camera.position.z = 500;

const rerender = () => {
    renderer.render(scene, camera);
}
rerender();

window.addEventListener('keydown', e => {
    if(e.key === 'ArrowLeft') sprite.position.x -= 10;
    if(e.key === 'ArrowRight') sprite.position.x += 10;
    rerender();
})
```

After some code-digging, I found that the problem seems to be Frustrum's .intersectsSprite method, which ignores the Sprite's .center property entirely.

Seems like replacing `_sphere.center.set( 0, 0, 0 )` with `_sphere.center.set( 0.5 - sprite.center.x, 0.5 - sprite.center.y, 0 )` would fix the issue?

**Comment by WestLangley** [comment]
Yes, `Frustum.intersectSprite()` does not honor the sprite's center.

In the mean time, you can try `sprite.frustumCulled = false`.

**Comment by WestLangley** [comment]
/ping @06wj Any interest in trying to fix this?

**Comment by picacure** [comment]
@06wj good boy

**Comment by dcm03** [comment]
Yes, it appears that the THREE.Frustum.intersectsSprite method does not take into account the center property of the THREE. Sprite when determining whether the sprite intersects the frustum. As a result, when the sprite moves towards the edge of the frustum, it may be considered to be outside of the frustum and not be rendered even if a significant portion of the sprite is still within the frustum.

To fix this issue, you could try modifying the intersectsSprite method as you suggested, by replacing _sphere.center.set( 0, 0, 0 ) with _sphere.center.set( 0.5 - sprite.center.x, 0.5 - sprite.center.y, 0 ). This should set the center of the bounding sphere to the center of the sprite, taking into account the center property of the sprite.

Alternatively, you could try using a different method for determining whether the sprite intersects the frustum. For example, you could use the intersectsBox method of THREE.Frustum, passing in a THREE.Box3 that is centered at the position of the sprite and has dimensions equal to the size of the sprite. This would take into account the full size and position of the sprite, including the center property.

It's worth noting that modifying the intersectsSprite method or using a different method for determining intersections may have unintended consequences and may not work in all cases. It's always a good idea to thoroughly test any changes to ensure that they are working as expected.
